casepro/utils/__init__.py

```diff
@@ -59,6 +60,23 @@ def json_encode(data):
     return json.dumps(data, cls=JSONEncoder)
 
 
+def json_decode(data):
+    """
+    Decodes the given JSON as primitives
+    """
+    if isinstance(data, six.binary_type):
+        data = data.decode('utf-8')
```

Comment:
Unfortunately we don't send the encoding in the header, but it is possible to add it. The http requests for Junebug are handled here: https://github.com/praekelt/junebug/blob/develop/junebug/workers.py#L245-L249

If I'm remembering correctly, `json.dumps` creates ascii strings with `\uxxxx` strings for unicode characters.